docs/UPGRADING.md

```diff
@@ -84,6 +84,18 @@ Areas where the changes are needed:
 
 - Instead of referring the `TaskExecutor` instance in the `map` function, you need to switch to the `ExeUnit` instance provided by the `ResourceRental`
 
+> **IMPORTANT DIFFERENCE**
+>
+> `TaskExecutor` by default assumed that users want to run these commands in parallel, so it
+> was establishing up to 5 agreements with different providers and then sent out these
+> commands to different providers to leverage parallelism.
+>
+> With `golem-js@3.0`, when you use `oneOf` to rent resources, you are establishing one agreement
+> with a single provider and the commands which you issue with `Array.map` will be executed
+> sequentially by that single provider.
+>
+> If you want to achieve _parallel command execution_ on different providers, refer to the [Engaging with many providers at the same time](#engaging-with-many-providers-at-the-same-time) example.
+
```

Comment:
Why even show this in the migration if it's fundamentally different behavior? TE users only care about parallelism. Either drop the entire `Repeating commands with map of different parameter values` section or use `manyOf` here

Reply:
Hmmmm. I guess you're right, this is more an example of the API usage than real migration. Thanks for that point of view :)